core/lib/presentation/views/dialog/confirmation_dialog_builder.dart

```diff
@@ -109,27 +123,38 @@ class ConfirmDialogBuilder {
   }
 
   Widget build() {
-    return Dialog(
-      key: _key,
-      shape: RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16))),
-      insetPadding: EdgeInsets.symmetric(horizontal: 24.0, vertical: 16.0),
-      child: Container(
-        margin: EdgeInsets.zero,
-        padding: EdgeInsets.zero,
-        width: 400,
+    if (showAsBottomSheet) {
+      return _bodyContent();
+    } else {
+      return Dialog(
+        key: _key,
+        shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(16))),
+        insetPadding: EdgeInsets.symmetric(
+            horizontal: 24.0,
+            vertical: 16.0),
+        child: _bodyContent(),
+      );
+    }
+  }
+
+  Widget _bodyContent() {
+    return Container(
+        width: _widthDialog ?? 400,
         decoration: BoxDecoration(
             color: Colors.white,
             borderRadius: BorderRadius.all(Radius.circular(16))),
+        margin: _margin,
         child: Wrap(children: [
           if (_onCloseButtonAction != null)
             Align(
-              alignment: Alignment.centerRight,
-              child: Padding(
-                padding: EdgeInsets.only(top: 8, right: 8),
-                child: buildIconWeb(
-                    icon: SvgPicture.asset(_imagePath.icCloseMailbox, fit: BoxFit.fill),
-                    onTap: () => _onCloseButtonAction?.call())
-              )),
+                alignment: Alignment.centerRight,
+                child: Padding(
+                    padding: EdgeInsets.only(top: 8, right: 8),
```

Comment:
can use const here

Reply:
@sherlockvn Can you write a sample for me?